magento2-varnish/4.0/etc/confd/templates/varnish/default.vcl.tmpl

```diff
@@ -17,6 +24,14 @@ acl purge {
 }
 
 sub vcl_recv {
+{{ if eq "true" (getenv "MAGENTO_USE_SEPARATE_ADMIN_CONTAINER") }}
+    if (req.url ~ "^/(index.php/)?{{ getenv "MAGENTO_ADMIN_FRONTNAME_REGEX_ESCAPED" }}/|/static/.*/adminhtml/Magento/backend/.*requirejs.*{{ if eq "true" (getenv "MAGENTO_USE_ADMIN_CONTAINER_FOR_MEDIA") }}|/media{{ end }}") {
```

Comment:
the `^` is only applying to the first url, the /static/ and /media/ are unanchored I believe